router/functionHandler.go

```diff
@@ -305,3 +308,34 @@ func (fh *functionHandler) handler(responseWriter http.ResponseWriter, request *
 
 	proxy.ServeHTTP(responseWriter, request)
 }
+
+// addForwardedHostHeader add "forwarded host" to request header
+func addForwardedHostHeader(req *http.Request) {
+	// for more detailed information, please visit:
+	// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded
+
+	var isIPv6 bool
+	var host string
+
+	// check whether a host is ipv4 or ipv6 or FQDN
+	ip := net.ParseIP(req.Host)
+	// The order here matters, To16() converts an IPv4 address
+	// to IPv6 format address. To prevent accidentally append
+	// IPv4 address to header need to check whether To4() is nil first.
+	if ip.To4() != nil {
+		isIPv6 = false
+	} else if v := ip.To16(); v != nil {
+		isIPv6 = true
+	}
+
+	if isIPv6 {
+		// For the "Forwarded" header, if a host is an IPv6 address
+		// it should be quoted and enclosed in square brackets
+		host = fmt.Sprintf("\"[%v]\"", req.Host)
+	} else {
+		host = req.Host
+	}
+
+	req.Header.Set("Forwarded", "host="+host)
+	req.Header.Set("X-Forwarded-Host", req.Host)
```

Comment:
Don't add this header if it already exists.

Reply:
(Both headers I mean. I think existing Forwarded/XFH headers should just be passed on unmodified... as far as I can understand.)